contract/models/abstract_contract.py

```diff
@@ -0,0 +1,72 @@
+# Copyright 2004-2010 OpenERP SA
+# Copyright 2014 Angel Moya <angel.moya@domatix.com>
+# Copyright 2015-2020 Tecnativa - Pedro M. Baeza
+# Copyright 2016-2018 Carlos Dauden <carlos.dauden@tecnativa.com>
+# Copyright 2016-2017 LasLabs Inc.
+# Copyright 2018 ACSONE SA/NV
+# License AGPL-3.0 or later (http://www.gnu.org/licenses/agpl).
+
+from odoo import api, fields, models
+
+
+class ContractAbstractContract(models.AbstractModel):
+    _inherit = "contract.recurrency.basic.mixin"
+    _name = "contract.abstract.contract"
+    _description = "Abstract Recurring Contract"
+
+    # These fields will not be synced to the contract
+    NO_SYNC = ["name", "partner_id", "company_id"]
+
+    name = fields.Char(required=True)
+    # Needed for avoiding errors on several inherited behaviors
+    partner_id = fields.Many2one(
+        comodel_name="res.partner", string="Partner", index=True
+    )
+    pricelist_id = fields.Many2one(comodel_name="product.pricelist", string="Pricelist")
+    contract_type = fields.Selection(
+        selection=[("sale", "Customer"), ("purchase", "Supplier")],
+        default="sale",
+        index=True,
+    )
+
+    journal_id = fields.Many2one(
+        "account.journal",
+        string="Journal",
+        default=lambda s: s._default_journal(),
+        domain="[('type', '=', contract_type)," "('company_id', '=', company_id)]",
+        index=True,
+    )
+    company_id = fields.Many2one(
+        "res.company",
+        string="Company",
+        required=True,
+        default=lambda self: self.env.company.id,
+    )
+    line_recurrency = fields.Boolean(
```

Comment:
IMO, it's a company strategical choice more than a user decision on individual contract. I was imagining it as a company config.

Reply:
I don't think so, as you can handle both type of contracts, or change them across time. Example:

- You start with a contract with 2 lines invoiced yearly.
- You create the contract without checking the line recurrency, and create easily with popups those lines.
- After 3 years, you need to add a monthly service. Then you check line, and define that third line through the popup, having the other two with previous recurrency data.

For company default configuration, just define a global default value for this field.